pkg/injector/config.go

```diff
@@ -14,6 +14,7 @@ type Config struct {
 	SidecarImage           string `envconfig:"SIDECAR_IMAGE" required:"true"`
 	SidecarImagePullPolicy string `envconfig:"SIDECAR_IMAGE_PULL_POLICY"`
 	Namespace              string `envconfig:"NAMESPACE" required:"true"`
+	KubeClusterDomain      string `envconfig:"KUBE_CLUSTER_DOMAIN"`
```

Comment:
It would be great if we could avoid adding K8s specific stuff here. Is there a way to auto-detect the k8s cluster domain in the code?

Reply:
I think there are two method to auto-detect:
1.  net.lookCNAME("kubernetes.default.svc") and grep the result get clusterDomain,such as:
```
/ # nslookup kubernetes.default.svc
Server:    10.96.0.10
Address 1: 10.96.0.10 kube-dns.kube-system.svc.cluster.local

Name:      kubernetes.default.svc
Address 1: 10.96.0.1 kubernetes.default.svc.cluster.local

```
2. grep /etc/resolv.conf file,get the "search" key word and get clusterDomain value
```
/ # cat /etc/resolv.conf
nameserver 10.96.0.10
search default.svc.cluster.local svc.cluster.local cluster.local
options ndots:5
```
but there have some problem,when Pod use Hostnetwork this way can not work beacuse resolv.conf use host value
```
/ # cat /etc/resolv.conf
nameserver 192.168.0.1

/ # nslookup kubernetes.default.svc
Server:    192.168.0.61
Address 1: 192.168.0.61

nslookup: can't resolve 'kubernetes.default.svc'

```

Reply:
Hi @artursouza i have push the code for auto-detect the k8s cluster domain